### ISSUE TITLE
determinize-prow-jobs invalidates TEST_SKIP

### ISSUE THREAD

**Issue by mjudeikis** [openshift/ci-operator#150]
We trying to skip some invalid/broken/not-relevant tests for `openshift-azure`. But we have failure in `determinize-prow-jobs`:
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/1520/pull-ci-openshift-release-master-ordered-prow-config/544

But if we commit with "determinized" content we get invalid regexp:
```
.*((The HAProxy router should set Forwarded headers appropriately)|(oc            new-app should succeed with a --name of 58 characters)|(process valueFrom            in build strategy environment variables should successfully resolve valueFrom            in docker build environment variables)|(prune builds based on settings            in the buildconfig should prune failed builds based on the failedBuildsHistoryLimit            setting)|(Prometheus when installed to the cluster should start and expose            a secured proxy and unsecured metrics)|(Prometheus when installed to the            cluster should start and expose a secured proxy and verify build metrics)|(templateservicebroker)|(deploymentconfigs            with multiple image change triggers [Conformance] should run a successful            deployment with multiple triggers)|(templateinstance readiness test should            report failed soon after an annotated objects has failed)|(s2i build with            a root user image should create a root build and pass with a privileged            SCC)).*
```
Which has a lot of whitespaces and skips are ignored.

Related to https://github.com/openshift/release/pull/1341
cc: @kargakis @stevekuznetsov @wking

**Comment by wking** [comment]
>    ...(deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with multiple triggers)...

Those brackets probably need to be escaped.  In my openshift/release#1341 script, see:

     processed = re.escape(''.join(chars).strip()) processed = processed.replace(r'\ ', ' ').replace(r'\-', '-').replace(r'\/', '/').replace(r'\=', '=') # Work around https://bugs.python.org/issue29995, fixed in Python 3.7

Dunno if that's what's giving you the error message.  You could try dropping `|(...)` blocks until [`MustCompile`][1] accepts it.

[1]: https://golang.org/pkg/regexp/#MustCompile

**Comment by mjudeikis** [comment]
@wking you right for the exit of the brackets. But this should invalidate only one test skipping:
https://play.golang.org/p/ZgcudECvkWT

But this, this might not be an issue.
